assets/src/stories-editor/components/with-wrapper-props.js

```diff
@@ -48,10 +48,13 @@ const withWrapperProps = ( BlockListBlock ) => {
 			return <BlockListBlock { ...props } wrapperProps={ wrapperProps } />;
 		}
 
+		const noCaption = ( 'core/image' === blockName && ! attributes.ampShowImageCaption ) ||
+			( 'core/video' === blockName && ! attributes.ampShowCaption );
+
 		// If we have image caption or font-family set, add these to wrapper properties.
 		wrapperProps = {
 			...props.wrapperProps,
-			'data-amp-image-caption': ( 'core/image' === blockName && ! attributes.ampShowImageCaption ) ? 'noCaption' : undefined,
+			'data-amp-caption': noCaption ? 'noCaption' : undefined,
```

Comment:
Does this affect block validation?

Reply:
Nope, this is in the editor only. In the FE an empty caption isn't displayed anyway.